### DISCUSSION TITLE
How to create different middleware for route groups?

### DISCUSSION THREAD

**Discussion by HaceraI** [vercel/next.js#62970] [Help] [answered]
### Summary

In NextJS14, the `middleware` file needs to be placed at the same level as the `app` and `pages` directories to take effect. So how should I configure `middleware` for different routing groups?
The routing group will not be reflected in the URL. 
Do I need to change the middleware file every time I add a new route?
![image](https://github.com/vercel/next.js/assets/48341368/861a957d-cca3-4362-b5a6-e82766c864a7)

### Additional information

```js
next version: 14.1.2
```

**Comment by HaceraI** [comment]
How to make middleware work only for `(main)` routing group? (I don't want to change this file every time the (basic) routing group changes)

This is my `middleware.ts`
```typescript
import { NextResponse } from 'next/server';
import type { NextRequest } from 'next/server';

export function middleware(request: NextRequest) {
  const isLoggedIn = checkUserLoggedIn(request);

  if (!isLoggedIn) {
    return NextResponse.redirect(new URL('/login', request.url));
  }

  return NextResponse.next();
}

function checkUserLoggedIn(request: NextRequest): boolean {
  return false;
}

```

**Comment by AmrViernes** [comment, marked as answer]
you need to chain middlewares this video gonna solve your problem
https://www.youtube.com/watch?v=fmFYH_Xu3d0&ab_channel=HamedBahram

**Reply by HaceraI** [reply]
ok, thanks

**Reply by atcheri** [reply]
sadly this doesn't seem to answer the question.
The video doesn't explain how to apply a middleware to a specific group.
It's mostly about chaining middlewares.
